apps/next/middleware.ts

```diff
@@ -1,18 +1,21 @@
-import { authMiddleware } from "@clerk/nextjs";
+import { clerkMiddleware, createRouteMatcher } from "@clerk/nextjs/server";
 
-// See https://clerk.com/docs/references/nextjs/auth-middleware
-// for more information about configuring your Middleware
-export default authMiddleware({
-  // Allow signed out users to access the specified routes:
-  // publicRoutes: ['/anyone-can-visit-this-route'],
-  publicRoutes: [
-    "/",
-    "/register",
-    "/login",
-    "/login-to-cli",
-    "/api/doc",
-    "/__nextjs_original-stack-frame",
-  ],
+// For more info, see:
+// https://clerk.com/docs/upgrade-guides/core-2/nextjs#migrating-to-clerk-middleware
+const isPublicRoute = createRouteMatcher([
+  // Allow signed out users to access these routes:
+  "/",
+  "/register",
+  "/login",
+  "/login-to-cli",
+  "/api/doc",
+  "/__nextjs_original-stack-frame",
+]);
+
+export default clerkMiddleware((auth, request) => {
+  if (!isPublicRoute(request)) {
```

Comment:
Yes that's an option, but doing that means a route is public until we _explicitly_ state that it's private. Instead, my current implementation treats any route as private until we state otherwise. I think private-by-default is better. Another consideration is that as the app grows the number of private routes will likely grow much larger than the number of public routes, so the suggested matcher will end up having a lot of routes, and that won't be easy to manage. Lastly, using a matcher on private routes means any member of the team that creates a new route/page meant for users only has to always remember to update the said list- that's cognitive load that I don't think we need.

However if you feel strongly about it, I can change it as suggested.

Reply:
ok sure it's fine as is